A regex pattern problem in PointSettings, in Android

In Android, the client seems to be unable to instantiate the PointSettings class, possibly due to an error in the Pattern regex at this line : 

private static final Pattern ENV_PROPERTY = Pattern.compile("(\\$\\{env\\.)(.+)(})");

----- begin exception -----
I/TestRunner: java.lang.NoClassDefFoundError: com.influxdb.client.write.PointSettings
        at com.influxdb.client.InfluxDBClientOptions$Builder.<init>(InfluxDBClientOptions.java:220)
        at com.influxdb.client.InfluxDBClientOptions.builder(InfluxDBClientOptions.java:199)
        at com.influxdb.client.InfluxDBClientFactory.create(InfluxDBClientFactory.java:108)
        at in.silentclap.sixpetal.facade.DbMethods.<init>(DbMethods.java:72)
        at in.silentclap.sixpetal.ExampleInstrumentedTest.checkConnection(ExampleInstrumentedTest.java:38)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at android.support.test.runner.AndroidJUnit4.run(AndroidJUnit4.java:101)
        at org.junit.runners.Suite.runChild(Suite.java:128)
        at org.junit.runners.Suite.runChild(Suite.java:27)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
        at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
        at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
        at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:384)
        at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2189)
     Caused by: java.lang.ExceptionInInitializerError
        at com.influxdb.client.InfluxDBClientOptions$Builder.<init>(InfluxDBClientOptions.java:220)
        at com.influxdb.client.InfluxDBClientOptions.builder(InfluxDBClientOptions.java:199)
        at com.influxdb.client.InfluxDBClientFactory.create(InfluxDBClientFactory.java:108)
        at in.silentclap.sixpetal.facade.DbMethods.<init>(DbMethods.java:72)
        at in.silentclap.sixpetal.ExampleInstrumentedTest.runOnceAfterDBCreation(ExampleInstrumentedTest.java:32)
        at java.lang.reflect.Method.invoke(Native Method) 
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) 
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) 
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) 
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) 
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325) 
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78) 
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57) 
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) 
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) 
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) 
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) 
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) 
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363) 
        at android.support.test.runner.AndroidJUnit4.run(AndroidJUnit4.java:101) 
        at org.junit.runners.Suite.runChild(Suite.java:128) 
        at org.junit.runners.Suite.runChild(Suite.java:27) 
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290) 
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71) 
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288) 
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58) 
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268) 
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363) 
        at org.junit.runner.JUnitCore.run(JUnitCore.java:137) 
        at org.junit.runner.JUnitCore.run(JUnitCore.java:115) 
        at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:56) 
        at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:384) 
        at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2189) 
     Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 17
    (\$\{env\.)(.+)(})
                     ^
        at java.util.regex.Pattern.compileImpl(Native Method)
        at java.util.regex.Pattern.compile(Pattern.java:1433)
        at java.util.regex.Pattern.<init>(Pattern.java:1408)
        at java.util.regex.Pattern.compile(Pattern.java:966)
        at com.influxdb.client.write.PointSettings.<clinit>(PointSettings.java:42)
        	... 33 more
    ----- end exception -----

--- a/client/src/main/java/com/influxdb/client/write/PointSettings.java
+++ b/client/src/main/java/com/influxdb/client/write/PointSettings.java
@@ -37,10 +37,12 @@ import com.influxdb.Arguments;
  *
  * @author Jakub Bednar (bednar@github) (28/06/2019 09:06)
  */
+@SuppressWarnings("RegExpRedundantEscape")
 public final class PointSettings {
 
-    private static final Pattern ENV_PROPERTY = Pattern.compile("(\\$\\{env\\.)(.+)(})");
-    private static final Pattern SYSTEM_PROPERTY = Pattern.compile("(\\$\\{)(.+)(})");
+    // Android compiler compatibility
+    private static final Pattern ENV_PROPERTY = Pattern.compile("(\\$\\{env\\.)(.+)\\}");
+    private static final Pattern SYSTEM_PROPERTY = Pattern.compile("(\\$\\{)(.+)(\\})");
 
     private final Map<String, String> defaultTags = new TreeMap<>();
 

